src/vs/workbench/contrib/snippets/browser/snippetCompletionProvider.ts

```diff
@@ -104,13 +104,17 @@ export class SnippetCompletionProvider implements CompletionItemProvider {
 				}
 			}
 
+			const word = model.getWordAtPosition(position);
 			let availableSnippets = new Set<Snippet>();
 			snippets.forEach(availableSnippets.add, availableSnippets);
 			suggestions = [];
 			for (let start of lineOffsets) {
 				availableSnippets.forEach(snippet => {
 					if (isPatternInWord(linePrefixLow, start, linePrefixLow.length, snippet.prefixLow, 0, snippet.prefixLow.length)) {
-						suggestions.push(new SnippetCompletion(snippet, Range.fromPositions(position.delta(0, -(linePrefixLow.length - start)), position)));
+						const replace = Range.fromPositions(position.delta(0, -(linePrefixLow.length - start)), { lineNumber: position.lineNumber, column: word ? word.endColumn : position.column });
+						const insert = replace.setEndPosition(position.lineNumber, position.column);
+
```

Comment:
Snippets completion doesn't stick to words and I believe the replace range needs to be derived from the prefix - e.g that part of the prefix that isn't "consumed" yet. Having this snippet `not word` and the prefix `no|t word` the the *insert* part should select `no` and the replace place should select `not word`

Reply:
@jrieken  What would be the expected behavior for the following cases with the snippet `not word`?
- `no|t wordmoreletters foo` => should the replace range be just `not word` or  `not wordmoreletters`
- `no|t wort foo` => should the replace range be just `not wor` or  `not wort`

Reply:
Yeah, that the question ;-) I'd say it should only replace what matches and insert otherwise, e.g. `no|t wordmoreletters foo` should become `not wordmoreletters` and `no|t wort foo` should become `not wordt wort foo`